### ISSUE TITLE
GrumPHP + phpCS = Problem with a large amount of files

### ISSUE THREAD

**Issue by Numline1** [phpro/grumphp#487]
| Q               | A
| --------------- | ---
| Version         | `GrumPHP 0.14.0`
| Bug?            | yes
| New feature?    |no
| Question?       | yes
| Documentation?  |no
| Related tickets | N/A

Hi! We've recently encountered a strange issue, where GrumPHP started failing with no prior changes
to it's version or our configuration. We run GrumPHP on our GitLab CI runner before each branch is merged into master. Only checks used are phpCS and phpUnit. phpCS started failing suddenly with no errors in the error output (code was clean and OK, again, no changes to it). I was able to trace down the issue and it appears that GrumPHP lists a large amount of files for phpCS to scan. Since Grum runs on the CI runner, it's not hooked to Git in any way, it just runs grumphp run and that's it. Anyway, as I wa saying, GrumPHP lists a large amount of files for phpCS to scan. Most of them were stripped away by the ignore_patterns setting, however, it still ended up listing several hundreds of them. I couldn't reproduce this issue locally (macOS High Sierra), but the Debian runner was apparently hitting some sort of POSIX argument limit. There was no error message even with -vvv switch, it just marked phpCS task as failed, hence blocking the entire merge. I managed to fix it by removing ignore_patterns and adding whitelist, which may have eliminated only about 10% of the files, however, now it runs okay.

What I'm basically proposing here is:

1) Does GrumPHP scan all the files when executed by `grumphp run` or is it just the latest commit? I presume it's all of them.
2) When executed directly like this, is it necessary to list all the scraped files one by one, instead of just executing phpcs --someargument --someothersetting . ? The "dot" will achieve the same thing.
3) Was this a different kind of issue I'm missing?

**My configuration**
```yaml
# grumphp.yml
    git_dir: .
    bin_dir: vendor/bin
    process_timeout: 600
    tasks:
        phpcs:
            standard: PSR2
            ignore_patterns: [./app/Http/Routes/*]
            whitelist_patterns: [/^app\/(.*)/]
            encoding: UTF-8
        phpunit:

```

PS: The original configuration was slightly different from the above, it listed more ignore_patterns and no whitelist_patterns, we were taking the approach of blacklisting instead of whitelisting. This configuration works just fine, the previous was the one hitting some sort of limit.

**Comment by veewee** [comment]
By default, grumphp adds all the files which are registered in git to the task.
This could be fixed by not adding the files in `RunContext`.
However, this will result in a broken `whitelist` configuration parameter since this is not an option on the actual cli tool.

**Comment by veewee** [comment]
Fixed in #796
